atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraNamespaceDeleterTest.java

```diff
@@ -0,0 +1,186 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import static com.palantir.logsafe.testing.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.cassandra.ImmutableCassandraCredentialsConfig;
+import com.palantir.atlasdb.cassandra.ImmutableCassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.namespacedeleter.NamespaceDeleter;
+import java.util.List;
+import java.util.Map;
+import org.apache.cassandra.thrift.KsDef;
+import org.apache.cassandra.thrift.NotFoundException;
+import org.apache.cassandra.thrift.TimedOutException;
+import org.apache.thrift.TException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class CassandraNamespaceDeleterTest {
+    private static final String KEYSPACE = "nonreservedkeyspace";
+    private static final ImmutableCqlQuery DROP_KEYSPACE_QUERY = CqlQuery.builder()
+            .safeQueryFormat(SchemaBuilder.dropKeyspace("\"" + KEYSPACE + "\"")
+                    .ifExists()
+                    .buildInternal())
+            .build();
+
+    @Mock
+    private CassandraClient cassandraClient;
+
+    private NamespaceDeleter namespaceDeleter;
+
+    @Before
+    public void before() {
+        namespaceDeleter = new CassandraNamespaceDeleter(getConfig(), () -> cassandraClient);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsKeyspaceSpecifiedInConfig() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        verify(cassandraClient).execute_cql3_query(eq(DROP_KEYSPACE_QUERY), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsOnlyKeyspaceSpecified() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        verify(cassandraClient, times(1)).execute_cql3_query(any(), any(), any());
```

Comment:
what does this test actually achieve over the above? I feel like if the above test added `verifyNoMoreInteractions` (or some other clever `never`-based thing, seeing as you will check the schema versions) it'd encapsulate it better - this test only confirms that a keyspace is dropped, not that it is the one specified.

Reply:
I was trying to work around the lack of noMoreInteractions for a given method - I did find a suitable not matcher though!